setup.py

```diff
@@ -66,69 +73,62 @@ def main():
+        print('Installed ROS2 deps, exiting...')
         return 0
 
     # Check CUDA
-    def is_cuda_version_ok():
-        nvcc_process = subprocess.run("nvcc --version", shell=True, stdout=subprocess.PIPE)
-        nvcc_ver_match = re.search("V[0-9]+.[0-9]+.[0-9]+", nvcc_process.stdout.decode("utf-8"))
-        if not nvcc_ver_match:
-            raise RuntimeError("CUDA not found")
-        major = int(nvcc_ver_match[0].split(".")[0][1:])  # [1:] to remove char 'v'
-        minor = int(nvcc_ver_match[0].split(".")[1])
-        patch = int(nvcc_ver_match[0].split(".")[2])
-        print(f"Found CUDA {major}.{minor}.{patch}")
-        for (actual, expected) in [(major, cfg.CUDA_MIN_VER_MAJOR), (minor, cfg.CUDA_MIN_VER_MINOR), (patch, cfg.CUDA_MIN_VER_PATCH)]:
-            if actual > expected:
-                return True
-            if actual < expected:
-                return False
-        return True
-
-    if not is_cuda_version_ok():
-        raise RuntimeError(f"CUDA version not supported! Get CUDA {cfg.CUDA_MIN_VER_MAJOR}.{cfg.CUDA_MIN_VER_MINOR}.{cfg.CUDA_MIN_VER_PATCH}+")
+    if not is_cuda_version_ok(cfg):
+        raise RuntimeError(
+            f"CUDA version not supported! Get CUDA {cfg.CUDA_MIN_VER_MAJOR}.{cfg.CUDA_MIN_VER_MINOR}.{cfg.CUDA_MIN_VER_PATCH}+")
 
     # Check OptiX_INSTALL_DIR
     if os.environ["OptiX_INSTALL_DIR"] == "":
         raise RuntimeError("OptiX not found! Make sure you have exported environment variable OptiX_INSTALL_DIR")
 
     # Check extension requirements
-    if args.with_pcl and not os.path.isdir(cfg.VCPKG_INSTALL_DIR):
-        raise RuntimeError("PCL extension requires dependencies to be installed: run this script with --install-pcl-deps flag")
+    if args.with_pcl and not os.path.isdir(cfg.VCPKG_DIR):
+        raise RuntimeError(
+            "PCL extension requires dependencies to be installed: run this script with --install-pcl-deps flag")
 
-    # Go to script directory
-    os.chdir(sys.path[0])
-
-    # Prepare build directory
-    if args.clean_build and os.path.isdir(args.build_dir):
-        shutil.rmtree(args.build_dir, ignore_errors=True)
-    if not os.path.isdir(args.build_dir):
-        os.makedirs(args.build_dir)
+    if args.with_ros2 and not os.path.isdir(cfg.RADAR_MSGS_INSTALL_DIR):
+        raise RuntimeError(
+            "ROS2 extension requires radar_msgs to be built: run this script with --install-ros2-deps flag")
 
     # Extend Path with libRobotecGPULidar location to link tests properly during the build on Windows
     if on_windows():
         os.environ["Path"] = os.environ["Path"] + ";" + os.path.join(os.getcwd(), args.build_dir)
 
+    if args.with_ros2:
+        # Source environment for additional packages
+        # ROS2 itself must be sourced by the user, because its location is unknown to this script
+        setup = "setup.bat" if on_windows() else "setup.sh"
+        source_environment(os.path.join(os.getcwd(), cfg.RADAR_MSGS_INSTALL_DIR, setup))
```

Comment:
I'd add a check if ROS2 is sourced at the beginning of this if-statement. It could produce a nicer output than this:
![image](https://github.com/RobotecAI/RobotecGPULidar/assets/112629916/e1bdfc37-fddc-4473-87f2-38690d950124)

Reply:
Done